lib/dialects/abstract/query-generator.js

```diff
@@ -291,8 +291,18 @@ const QueryGenerator = {
     let selectFromTmp = '';   // Select statement for trigger
 
     if (this._dialect.supports['LIMIT ON UPDATE'] && options.limit) {
-      if (this.dialect !== 'mssql') {
-        query += ' LIMIT ' + this.escape(options.limit) + ' ';
+      if (this.dialect === 'oracle') {
+        //This cannot be setted in where because rownum will be quoted
+        if (where && ((where.length && where.length > 0) || (Object.keys(where).length > 0))) {
+          //If we have a where clause, we add AND
+          query += ' AND ';
+        } else {
+          //No where clause, we add where
+          query += ' WHERE ';
+        }
+        query += `rownum < ${options.limit + 1}`;
```

Comment:
`options.limit` is not escaped, potential SQL injection vulnerability?

I recommend using `<=` over `<` to avoid the arithmetic and add self.escape (need to make sure it doesn't turn a number to a string). 

Also, the example below adds a space on the end (not sure if it's needed).

Maybe something like:

```javascript
query += `rownum <= this.escape(options.limit) `;
```